src/components/graph/network-modification-tree-model.js

```diff
@@ -19,9 +29,50 @@ export default class NetworkModificationTreeModel {
         this.treeEdges = [...this.treeEdges]; //otherwise react-flow doesn't show new edges
     }
 
-    addChild(newNode, parentId, insertMode) {
-        // Add node
-        this.treeNodes.push(convertNodetoReactFlowModelNode(newNode, parentId));
+    addChild(newNode, parentId, insertMode, referenceNodeId) {
+        // we have to keep a precise order of nodes in the array to avoid gettings children
+        // nodes before their parents when building graph in dagre library which have uncontrolled results
+        // We also need to do this to keep a correct order when inserting nodes and not loose the user.
+        const referenceNodeIndex = referenceNodeId
+            ? this.treeNodes.findIndex((node) => node.id === referenceNodeId)
+            : null;
+        switch (insertMode) {
+            case 'BEFORE': {
```

Comment:
create enum for "BEFORE", "AFTER", "CHILD"

Reply:
done